include/flang/Semantics/symbol.h

```diff
@@ -131,8 +131,18 @@ class AssocEntityDetails : public EntityDetails {
   AssocEntityDetails &operator=(AssocEntityDetails &&) = default;
   const MaybeExpr &expr() const { return expr_; }
 
+  void set_rank(int rank);
+  std::optional<int> associationRank() const {
+    if (associationRank_.has_value()) {
```

Comment:
Why not just `return associationRank_;`?

Reply:
I was thinking if in case the value of `std::optional `wouldn't be set checking the `has_value() ` would have been saving us from undefined behavior, but I found c++ compiler catches it with `std::bad_optional_access`. So anyways I have removed the guard.

Reply:
You can copy an `optional` whether it has a value or not without throwing an exception.